manifests/rvmrc.pp

```diff
@@ -1,11 +1,12 @@
 # Configure the /etc/rvmrc file
 class rvm::rvmrc (
-  $manage_group = $rvm::params::manage_group,
+  Boolean $manage_group = $rvm::params::manage_group,
   $template = 'rvm/rvmrc.erb',
-  $umask = 'u=rwx,g=rwx,o=rx',
-  $max_time_flag = undef,
-  $autoupdate_flag = 0,
-$silence_path_mismatch_check_flag = undef) inherits rvm::params {
+  String[0] $umask = 'u=rwx,g=rwx,o=rx',
+  Optional[Integer[0]] $max_time_flag = undef,
+  Enum['disabled', 'warn', 'enabled'] $autoupdate_flag = 'disabled',
```

Comment:
We can continue accepting `0`, `1` and `2` but I think just the named values provides a self-documenting experience.

Reply:
Can't we add both ways, just as rvm is doing it?

Reply:
We can, but it does make the data type more complicated. For example:
```suggestion
  Variant[Enum['disabled', 'warn', 'enabled'], Integer[0, 2]] $autoupdate_flag = 'disabled',
```

Or even:
```suggestion
  Variant[Enum['disabled', 'warn', 'enabled', '0', '1', '2'], Integer[0, 2]] $autoupdate_flag = 'disabled',
```

Personally I'm leaning to keeping it simple. The enum as it is in the commit now doesn't need documentation while the integers do.

Reply:
Thanks, then let's go with simple, as there are already other breaking changes.